structure-macro-impl/src/lib.rs

```diff
@@ -1,65 +1,71 @@
 #![recursion_limit = "128"]
 
-#[macro_use]
-extern crate proc_macro_hack;
-#[macro_use]
-extern crate quote;
+extern crate proc_macro;
+extern crate alloc;
 
-use std::mem;
-use std::os::raw::c_void;
-use std::string::String;
-use quote::{Tokens, Ident};
+use alloc::string::String;
+use core::mem;
+use core::ffi::c_void;
 
-proc_macro_expr_impl! {
-    pub fn structure_impl(input: &str) -> String {
-        let format = trim_quotes(input);
-        let struct_name = Ident::from(format_to_struct_name(format));
-        let (values, endianness) = format_to_values(&format);
-        let (args, fn_decl_args, args_types) = build_args_list(&values);
-        let endianness = match endianness {
-            Endianness::Native => {
-                if cfg!(target_endian = "little") {
-                    quote!(LittleEndian)
-                } else {
-                    quote!(BigEndian)
-                }
-            }
-            Endianness::LittleEndian => quote!(LittleEndian),
-            Endianness::BigEndian => quote!(BigEndian),
-        };
-        let size = calc_size(&values);
-        let pack_fn = build_pack_fn(&args, &fn_decl_args, size);
-        let pack_into_fn = build_pack_into_fn(&values, &fn_decl_args, &endianness);
-        let unpack_fn = build_unpack_fn(&args_types, size);
-        let unpack_from_fn = build_unpack_from_fn(&values, &args, &args_types, &endianness);
-        let size_fn = build_size_fn(size);
-        let output = quote! {{
-            #[derive(Debug)]
-            #[allow(non_camel_case_types)]
-            struct #struct_name;
-            #[allow(unused_imports)]
-            use std::io::{Result, Write, Read, Error, ErrorKind, Cursor};
-            #[allow(unused_imports)]
-            use std::os::raw::c_void;
-            #[allow(unused_imports)]
-            use structure::byteorder::{WriteBytesExt, ReadBytesExt, BigEndian, LittleEndian};
+#[allow(unused_imports)]
```

Comment:
If they are unused, remove them.

Reply:
You may need to force pull, but I've fixed this. Not sure why GH isn't reflecting that. I squashed the commit this was in, but you may want to see 865508c too.